core/src/test/java/org/elasticsearch/discovery/zen/UnicastZenPingTests.java

```diff
@@ -82,13 +90,13 @@
 import static org.elasticsearch.gateway.GatewayService.STATE_NOT_RECOVERED_BLOCK;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.hasSize;
 import static org.mockito.Matchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 
+@TestLogging("org.elasticsearch.transport:TRACE,org.elasticsearch.discovery.zen:TRACE")
```

Comment:
This logging was initially added to just `testSimplePings` to chase a race. The race has not reproduced since adding this logging. I think that we should drop the logging and and then address if the race comes back since you've changed how these things are handled.